### PR TITLE
fix: update actions

### PR DESCRIPTION
**Pull request by leomelzer**
- Update to checkout@v2
- Use caching mechanism from setup-node@v2
- Use node v16 instead of v15

### CODE REVIEW

**Comment by leomelzer** [comment]
I believe this PR isn't really relevant, the project isn't used (!?) Any longer. Closing now.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -4,23 +4,19 @@ on:
     branches:
       - master
 jobs:
-  prepare:
-    runs-on: ubuntu-latest
-    if: "! contains(github.event.head_commit.message, '[skip ci]')"
-    steps:
-      - run: echo "${{ github.event.head_commit.message }}"
```
Comment (jhnns) [inline .github/workflows/release.yml:11]: I'm wondering why you've removed this step? :) I think it's still necessary

**Comment by jhnns** [comment]
All good, the project is still used in our TypeScript workshop :)
I'll merge it as soon as the last thing is resolved.

File: .github/workflows/release.yml (hunk 1 above)
Comment (leomelzer) [reply .github/workflows/release.yml:11]: This is natively supported for some time now: https://github.blog/changelog/2021-02-08-github-actions-skip-pull-request-and-push-workflows-with-skip-ci/

That's why I think it isn't needed any longer :)

**Comment by leomelzer** [comment]
Thanks for reopening @jhnns!

File: .github/workflows/release.yml (hunk 1 above)
Comment (jhnns) [reply .github/workflows/release.yml:11]: Thanks for clarification. I didn't know that :)

**Review by jhnns — APPROVED** [review]

**Pull request merged**